File: library_core/src/main/java/cn/flyfun/ktx/gamesdk/core/impl/login/LoginFragmentPagerAdapter.java
package cn.flyfun.ktx.gamesdk.core.impl.login;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import cn.flyfun.ktx.gamesdk.core.impl.login.fragment.ChooseFragment;
import cn.flyfun.ktx.gamesdk.core.impl.login.fragment.RegisterFragment;

/**
 * @author #Suyghur.
 * Created on 10/30/20
 */
public class LoginFragmentPagerAdapter extends FragmentPagerAdapter {

    private String[] titles;

    private Fragment mCurrentFragment;

    public LoginFragmentPagerAdapter(String[] titles, FragmentManager fm) {
        super(fm);
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 1) {
            return new RegisterFragment();
        }
        return new ChooseFragment();
    }

    @Override
    public int getCount() {
        return titles.length;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles[position];
    }

    @Override
    public void setPrimaryItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        mCurrentFragment = (Fragment) object;
        super.setPrimaryItem(container, position, object);
    }

    public Fragment getCurrentFragment() {
        return mCurrentFragment;
    }
}

File: library_core/src/main/java/cn/flyfun/gamesdk/core/impl/login/LoginFragmentPagerAdapter.kt
package cn.flyfun.gamesdk.core.impl.login

import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import cn.flyfun.gamesdk.core.impl.login.fragment.ChooseFragment
import cn.flyfun.gamesdk.core.impl.login.fragment.RegisterFragment

/**
 * @author #Suyghur.
 * Created on 2020/12/10
 */
class LoginFragmentPagerAdapter constructor(private val titles: Array<String>, var fm: FragmentManager) : FragmentPagerAdapter(fm) {


    var currentFragment: Fragment? = null
        private set

    override fun getCount(): Int {
        return titles.size
    }

    override fun getItem(position: Int): Fragment {
        return if (position == 1) {
            RegisterFragment()
        } else {
            ChooseFragment()
        }
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titles[position]
    }

    override fun setPrimaryItem(container: ViewGroup, position: Int, `object`: Any) {
        currentFragment = `object` as Fragment
        super.setPrimaryItem(container, position, `object`)
    }

}